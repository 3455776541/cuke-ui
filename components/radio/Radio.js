import React, { PureComponent } from "react";
import cls from "classnames";
import PropTypes from "prop-types";

export default class Radio extends PureComponent {
  state = {
    checked: this.props.checked || this.props.defaultChecked || false
  };
  static defaultProps = {
    prefixCls: "cuke-radio",
    defaultChecked: false,
    checked: false
  };

  static propTypes = {
    prefixCls: PropTypes.string.isRequired,
    onChange: PropTypes.func,
    disabled: PropTypes.bool,
    checked: PropTypes.bool,
    defaultChecked: PropTypes.bool
  };

  constructor(props) {
    super(props);
  }

  onChange = e => {
    this.setState({
      checked: true
    });
    if (this.props.onChange) {
      this.props.onChange(e);
    }
  };

  componentWillReceiveProps({ checked, defaultChecked }) {
    if (checked !== this.state.checked) {
      this.setState({
        checked: defaultChecked || checked
      });
    }
  }

  render() {
    const {
      prefixCls,
      value,
      className,
      children,
      disabled,
      onChange, //eslint-disable-line
      ...attr
    } = this.props;

    const { checked } = this.state;

    return (
      <label className={cls(`${prefixCls}-wrapper`)} {...attr}>
        <span
          className={cls(prefixCls, className, {
            [`${prefixCls}-checked`]: checked,
            [`${prefixCls}-disabled`]: disabled
          })}
        >
          <input
            type="radio"
            value={value}
            checked={checked}
            className={cls(`${prefixCls}-input`)}
            onChange={this.onChange}
            disabled={disabled}
          />
          <span className={cls(`${prefixCls}-inner`)} />
        </span>
        <span>{children}</span>
      </label>
    );
  }
}
