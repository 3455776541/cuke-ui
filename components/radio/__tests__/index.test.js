import React from "react";
import { render, shallow } from "enzyme";
import toJson from "enzyme-to-json";
import Radio from "../index";

describe("<Radio/>", () => {
	it("should render Radio", () => {
		const wrapper = render(
			<div>
				<Radio>黄瓜 ui</Radio>
				<Radio disabled>黄瓜 ui</Radio>
				<Radio checked disabled />
			</div>
		);
		expect(toJson(wrapper)).toMatchSnapshot();
	});

	it("should render RadioGroup", () => {
		const wrapper = render(
			<div>
				<Radio.Group value={"小红"}>
					<Radio value="小红">小红</Radio>
					<Radio value="小明">小明</Radio>
					<Radio value="小美">小美</Radio>
					<Radio value="小芳">小芳</Radio>
					<Radio value="小黑" disabled>
						小黑
					</Radio>
				</Radio.Group>
			</div>
		);
		expect(toJson(wrapper)).toMatchSnapshot();
	});
	// it('should trigger change event width RadioGroup', () => {
	// 	const onChange = jest.fn();
	// 	const wrapper = shallow(
	// 		<Radio.Group value={'小红'} onChange={onChange}>
	// 			<Radio value="小红">小红</Radio>
	// 			<Radio value="小明">小明</Radio>
	// 			<Radio value="小美">小美</Radio>
	// 			<Radio value="小芳">小芳</Radio>
	// 		</Radio.Group>
	// 	);
	// 	wrapper.find('input').simulate('change');
	// 	expect(onChange).toHaveBeenCalled();
	// });
	it("should can trigger change event", () => {
		const onChange = jest.fn();
		const wrapper = shallow(<Radio onChange={onChange}>黄瓜 ui</Radio>);
		wrapper.find("input").simulate("change");
		expect(onChange).toHaveBeenCalled();
	});
	it("should can not trigger change event", () => {
		const onChange = jest.fn();
		const wrapper = shallow(
			<Radio onChange={onChange} disabled>
				黄瓜 ui
			</Radio>
		);
		wrapper.find("input").simulate("click");
		expect(onChange).not.toHaveBeenCalled();
	});
});
