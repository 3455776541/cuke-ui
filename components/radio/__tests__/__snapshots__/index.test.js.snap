// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Radio/> should render Radio 1`] = `
<div>
  <label
    class="cuke-radio-wrapper"
  >
    <span
      class="cuke-radio"
    >
      <input
        class="cuke-radio-input"
        type="radio"
      />
      <span
        class="cuke-radio-inner"
      />
    </span>
    <span>
      黄瓜 ui
    </span>
  </label>
  <label
    class="cuke-radio-wrapper cuke-radio-disabled"
  >
    <span
      class="cuke-radio"
    >
      <input
        class="cuke-radio-input"
        disabled=""
        type="radio"
      />
      <span
        class="cuke-radio-inner"
      />
    </span>
    <span>
      黄瓜 ui
    </span>
  </label>
  <label
    checked=""
    class="cuke-radio-wrapper cuke-radio-checked cuke-radio-disabled"
  >
    <span
      class="cuke-radio"
    >
      <input
        checked=""
        class="cuke-radio-input"
        disabled=""
        type="radio"
      />
      <span
        class="cuke-radio-inner"
      />
    </span>
    <span />
  </label>
</div>
`;

exports[`<Radio/> should render Radio.Button 1`] = `
<div
  class="cuke-radio-group"
>
  <label
    checked=""
    class="cuke-radio-wrapper cuke-radio-button-wrapper cuke-radio-checked cuke-radio-default"
  >
    <span
      class="cuke-radio cuke-radio-button"
    >
      <input
        checked=""
        class="cuke-radio-input"
        type="radio"
      />
      <span
        class="cuke-radio-inner"
      />
    </span>
    <span>
      黄瓜 ui
    </span>
  </label>
  <label
    checked=""
    class="cuke-radio-wrapper cuke-radio-button-wrapper cuke-radio-checked cuke-radio-disabled cuke-radio-default"
  >
    <span
      class="cuke-radio cuke-radio-button"
    >
      <input
        checked=""
        class="cuke-radio-input"
        disabled=""
        type="radio"
      />
      <span
        class="cuke-radio-inner"
      />
    </span>
    <span>
      黄瓜 ui
    </span>
  </label>
  <label
    checked=""
    class="cuke-radio-wrapper cuke-radio-button-wrapper cuke-radio-checked cuke-radio-disabled cuke-radio-default"
  >
    <span
      class="cuke-radio cuke-radio-button"
    >
      <input
        checked=""
        class="cuke-radio-input"
        disabled=""
        type="radio"
      />
      <span
        class="cuke-radio-inner"
      />
    </span>
    <span />
  </label>
</div>
`;

exports[`<Radio/> should render RadioGroup 1`] = `
<div>
  <div
    class="cuke-radio-group"
    value="小红"
  >
    <label
      checked=""
      class="cuke-radio-wrapper cuke-radio-checked"
    >
      <span
        class="cuke-radio"
      >
        <input
          checked=""
          class="cuke-radio-input"
          type="radio"
          value="小红"
        />
        <span
          class="cuke-radio-inner"
        />
      </span>
      <span>
        小红
      </span>
    </label>
    <label
      class="cuke-radio-wrapper"
    >
      <span
        class="cuke-radio"
      >
        <input
          class="cuke-radio-input"
          type="radio"
          value="小明"
        />
        <span
          class="cuke-radio-inner"
        />
      </span>
      <span>
        小明
      </span>
    </label>
    <label
      class="cuke-radio-wrapper"
    >
      <span
        class="cuke-radio"
      >
        <input
          class="cuke-radio-input"
          type="radio"
          value="小美"
        />
        <span
          class="cuke-radio-inner"
        />
      </span>
      <span>
        小美
      </span>
    </label>
    <label
      class="cuke-radio-wrapper"
    >
      <span
        class="cuke-radio"
      >
        <input
          class="cuke-radio-input"
          type="radio"
          value="小芳"
        />
        <span
          class="cuke-radio-inner"
        />
      </span>
      <span>
        小芳
      </span>
    </label>
    <label
      class="cuke-radio-wrapper cuke-radio-disabled"
    >
      <span
        class="cuke-radio"
      >
        <input
          class="cuke-radio-input"
          disabled=""
          type="radio"
          value="小黑"
        />
        <span
          class="cuke-radio-inner"
        />
      </span>
      <span>
        小黑
      </span>
    </label>
  </div>
</div>
`;

exports[`<Radio/> should render custom size 1`] = `
<div>
  <RadioGroup
    disabled={false}
    onChange={[Function]}
    prefixCls="cuke-radio-group"
    size="small"
    value="小红"
  >
    <Radio
      checked={false}
      defaultChecked={false}
      prefixCls="cuke-radio"
      size="default"
      value="小红"
    >
      小红
    </Radio>
    <Radio
      checked={false}
      defaultChecked={false}
      prefixCls="cuke-radio"
      size="default"
      value="小明"
    >
      小明
    </Radio>
    <Radio
      checked={false}
      defaultChecked={false}
      prefixCls="cuke-radio"
      size="default"
      value="小美"
    >
      小美
    </Radio>
  </RadioGroup>
  <RadioGroup
    disabled={false}
    onChange={[Function]}
    prefixCls="cuke-radio-group"
    size="default"
    value="小红"
  >
    <Radio
      checked={false}
      defaultChecked={false}
      prefixCls="cuke-radio"
      size="default"
      value="小红"
    >
      小红
    </Radio>
    <Radio
      checked={false}
      defaultChecked={false}
      prefixCls="cuke-radio"
      size="default"
      value="小明"
    >
      小明
    </Radio>
    <Radio
      checked={false}
      defaultChecked={false}
      prefixCls="cuke-radio"
      size="default"
      value="小美"
    >
      小美
    </Radio>
  </RadioGroup>
  <RadioGroup
    disabled={false}
    onChange={[Function]}
    prefixCls="cuke-radio-group"
    size="large"
    value="小红"
  >
    <Radio
      checked={false}
      defaultChecked={false}
      prefixCls="cuke-radio"
      size="default"
      value="小红"
    >
      小红
    </Radio>
    <Radio
      checked={false}
      defaultChecked={false}
      prefixCls="cuke-radio"
      size="default"
      value="小明"
    >
      小明
    </Radio>
    <Radio
      checked={false}
      defaultChecked={false}
      prefixCls="cuke-radio"
      size="default"
      value="小美"
    >
      小美
    </Radio>
  </RadioGroup>
</div>
`;
